internal/storage/v2/memory/factory.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package memory
+
+import (
+	"context"
+
+	"github.com/jaegertracing/jaeger/internal/distributedlock"
+	"github.com/jaegertracing/jaeger/internal/metrics"
+	"github.com/jaegertracing/jaeger/internal/storage/v1"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/api/samplingstore"
+	v1 "github.com/jaegertracing/jaeger/internal/storage/v1/memory"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/api/depstore"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore"
+	"github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore/tracestoremetrics"
+	"github.com/jaegertracing/jaeger/internal/telemetry"
+)
+
+var (
+	_ tracestore.Factory           = (*Factory)(nil)
+	_ storage.SamplingStoreFactory = (*Factory)(nil)
+	_ storage.Purger               = (*Factory)(nil)
+)
+
+type Factory struct {
+	store          *Store
+	metricsFactory metrics.Factory
+}
+
+func NewFactory(cfg v1.Configuration, settings telemetry.Settings) (*Factory, error) {
```

Comment:
out convention is to call this parameter `telset`
```suggestion
func NewFactory(cfg v1.Configuration, telset telemetry.Settings) (*Factory, error) {
```